main/provider/index.ts

```diff
@@ -399,13 +399,13 @@ export class Provider extends EventEmitter {
       const approvals: RequiredApproval[] = []
       const accountId = (accounts.current() || {}).id
       const rawTx = getRawTx(newTx, accountId)
-      const gas = gasFees(rawTx)
-      const chainConfig = this.connection.connections['ethereum'][parseInt(rawTx.chainId)].chainConfig
+      const [gas, updatedTx] = processTxForGasFees(rawTx)
```

Comment:
I wonder if this process can be streamlined. this flow was designed when were always taking the gas from the store, as it stands in this PR it's a bit circuitous where we take the gas settings from the tx, set them on the gas object, then read the gas object and set them back on the tx

Reply:
Yeah will look at this

Reply:
Definitely needs completely reworking, didn't realise this was the flow